src/check/arbitrary/IntegerArbitrary.ts

```diff
@@ -59,6 +59,7 @@ function integer(max: number): ArbitraryWithShrink<number>;
  */
 function integer(min: number, max: number): ArbitraryWithShrink<number>;
 function integer(a?: number, b?: number): ArbitraryWithShrink<number> {
+  if (a && b && a > b) throw new Error('fc.integer maximum value should be equal or greater than the minimum one');

```

Comment:
I'd recommend to check the following instead:
`a !== undefined && b !== undefined && a > b`

Current code has an issue if one of `a` or `b` is 0.
I found that with a property based test (see below).


Eg.: `a = 1` and `b = 0`

Reply:
Hi! In the new commit, I fixed the issue, but although I checked the code several times, I couldn't get why the property test was failing with `if (a && b && a > b) `but not with `if (a !== undefined && b !== undefined && a > b)`

Any ideas why that happened?

Reply:
Hi! The reason of the bug is that when `b = 0` the condition inside the `if` statement becomes falsy: `if (0) { /* never executed */ }`. So when we have `a = 1` and `b = 0` the statement is equivalent to `if (1 && 0 && 1 > 0)` which evaluates to `false` because of the `&& 0 &&`.